### DISCUSSION TITLE
how do we set last-modified?

### DISCUSSION THREAD

**Discussion by pass-techdev** [alexandercerutti/passkit-generator#229] [General]
Is there a way to set last-modified attribute in the manifest.json?

**Comment by alexandercerutti** [comment]
Hi @pass-techdev, thanks for using passkit-generator!

What is this `last-modified` attribute you are talking about? I never heard of it.
Furthermore, `manifest.json`? That is an autogenerated file...
If you have a link or a resource where I could look from, I'd be happy to give a look.

Let me know!

**Reply by pass-techdev** [reply]
it seems for the pass to be auto updatable, it needs to be in the manifest.json so likely will need to be included in headers then. Thanks for confirming.

**Reply by alexandercerutti** [reply]
Yeah, it is not like that.
`manifest.json` is a file containing the SHA-1 for all the files in the `.pkpass` except `manifest.json` itself and `signature`. It is required for validation of the pkpass integrity.

`last-modified` HTTP Header you are talking about is, indeed, an HTTP Header.
passkit-generator is the essential brick for generating a pkpass. It doesn't connect automatically to an HTTP Server.

To add an header, you need to develop an HTTP integration with Apple Wallet WebServices, for which I've developed three independent packages that help you setting them up and linking them to your current webserver integration (Fastify, Express, Hono.dev).

You can see all of them [here](https://github.com/alexandercerutti/passkit-webservice-toolkit) (read the README file).
All of them automatically handle the `last-modified` header for a matter of listing the passes to update.

You can try using them and perhaps ask in the repository of the project you are using if you have any further questions.

Let me know!
